contracts/src/CW721ERC721Pointer.sol

```diff
@@ -82,6 +92,26 @@ contract CW721ERC721Pointer is ERC721 {
         return string(uri);
     }
 
+    // 2981
+    function royaltyInfo(uint256 tokenId, uint256 salePrice) public view override returns (address, uint256) {
+        bytes memory checkRoyaltyResponse = WasmdPrecompile.query(Cw721Address, bytes("{\"extension\":{\"msg\":{\"check_royalties\":{}}}}"));
+        bytes memory isRoyaltyImplemented = JsonPrecompile.extractAsBytes(checkRoyaltyResponse, "royalty_payments");
+        if (keccak256(isRoyaltyImplemented) != keccak256("true")) {
+            revert("royalty info not implemented on the underlying CosmWasm contract");
+        }
+        string memory tId = _formatPayload("token_id", _doubleQuotes(Strings.toString(tokenId)));
+        string memory sPrice = _formatPayload("sale_price", _doubleQuotes(Strings.toString(salePrice)));
+        string memory req = _curlyBrace(_formatPayload("royalty_info", _curlyBrace(_join(tId, sPrice, ","))));
+        string memory fullReq = _curlyBrace(_formatPayload("extension", _curlyBrace(_formatPayload("msg", req))));
+        bytes memory response = WasmdPrecompile.query(Cw721Address, bytes(fullReq));
+        bytes memory addr = JsonPrecompile.extractAsBytes(response, "address");
+        uint256 amt = JsonPrecompile.extractAsUint256(response, "royalty_amount");
+        if (addr.length == 0) {
+            return (address(0), amt);
```

Comment:
should this be `(address(0), 0)`? I think we probably don't want to risk people burning funds on mistake.

Reply:
the cw2981 implementation makes look like it's a possible scenario for the payment address to be empty while the royalty is non-zero: https://github.com/CosmWasm/cw-nfts/blob/main/contracts/cw2981-royalties/src/lib.rs#L41-L45, so I'm not sure if we want to overwrite `amt` here?